src/main/java/com/strandls/cca/util/CCAUtil.java

```diff
@@ -107,4 +115,33 @@ public static MailData generateMailData(CCAData ccaData, String title, String de
 		}
 		return mailData;
 	}
+
+	public static List<UserGroupMailData> generateUserGroupMailData(UserGroupSerivceApi userGroupService,
+			Set<String> usergroups) {
+		List<UserGroupMailData> userGroupMailData = new ArrayList<>();
+		List<UserGroupIbp> updatedUG = new ArrayList<>();
+
+		for (String usergroupId : usergroups) {
+			UserGroupIbp userGroupIbp;
+			try {
+				userGroupIbp = userGroupService.getIbpData(usergroupId);
```

Comment:
This is correct and i updated the code , but to note that the above endpoint you shared seems kind of doing the same thing, multiple db calls.

Reply:
yes it  does,  but each REST api call has its own overhead apart from DB calls. Placing multilple api calls inside for loop only adds on to it.  Now ,even if there are multiple DB calls, rest api call will be only 1.
Ideal solution: edit the usergroup endpoint to get data in 1 DB call itself, it can utilise the same `IN` keyword in postregres we used once to optimise one such use case for media gallery feature.

Reply:
The above solution for optimisation(changing usergroup-api) can be taken up in the next iteration.